src/main/scala/com/ing/wbaa/airlock/proxy/provider/kafka/EventProducer.scala

```diff
@@ -0,0 +1,43 @@
+package com.ing.wbaa.airlock.proxy.provider.kafka
+
+import akka.Done
+import akka.stream.ActorMaterializer
+import com.ing.wbaa.airlock.proxy.config.KafkaSettings
+import org.apache.kafka.clients.producer.{ KafkaProducer, ProducerConfig, ProducerRecord }
+import org.apache.kafka.common.serialization.StringSerializer
+
+import scala.concurrent.duration.SECONDS
+import scala.concurrent.{ ExecutionContext, Future }
+import scala.util.{ Failure, Success, Try }
+
+trait EventProducer {
+  import scala.collection.JavaConverters._
+
+  protected[this] implicit val kafkaSettings: KafkaSettings
+
+  protected[this] implicit val materializer: ActorMaterializer
+
+  protected[this] implicit val executionContext: ExecutionContext
+
+  private lazy val config: Map[String, Object] =
+    Map[String, Object](
+      "bootstrap.servers" -> kafkaSettings.bootstrapServers,
+      ProducerConfig.RETRIES_CONFIG -> kafkaSettings.retries,
+      ProducerConfig.RECONNECT_BACKOFF_MS_CONFIG -> kafkaSettings.retriesBackOff,
+      ProducerConfig.RECONNECT_BACKOFF_MAX_MS_CONFIG -> kafkaSettings.retriesBackOffMax
+    )
+
+  def kafkaProducer: KafkaProducer[String, String] = new KafkaProducer(config.asJava, new StringSerializer, new StringSerializer)
+
+  def sendSingleMessage(event: String, topic: String): Future[Done] = {
+    val sentF = kafkaProducer.send(new ProducerRecord[String, String](topic, event))
+    Try(
+      sentF.get(3, SECONDS)
```

Comment:
Here the thread is blocked until the result is available. Instead of `Try`, I would use `Future` so that a different thread waits for the completion of the Java future and handles the exception too, e.g.
```scala
Future {
  sentF.get(3, SECONDS)
} transform {
  case Success(_) =>
    kafkaProducer.close()
    Success(Done)
  case Failure(ex) =>
    Failure(new Exception(s"Failed to send event to Kafka, ${ex.getMessage}"))
}
```